duckduckgo/iOS
# Update privacy toggle to work on a per site basis

Task/Issue URL: https://app.asana.com/0/72649045549333/559623300949084
Tech Design URL: N/A
CC:

**Description**:
Update privacy toggle to work on a per site basis.

**Steps to test this PR**:

**TEST 1: Test new toggle text and behavior**
1. Navigate to a page e.g cnn.com
1. Open the privacy dashboard and ensure that the toggle text is "Site Privacy Protection"
1. Switch privacy protection off and press _Done_
1. Ensure page reloads
1. Ensure page is now in whitelist
1. Ensure that mp_wla pixel is fired
1. Navigate to another page e.g facebook.com
1. Check that privacy toggle is on
1. Navigate back cnn.com and ensure that the privacy/site protection toggle is back off
1. Switch the toggle back on
1. Ensure that  mp_wlr pixel is fired
1. Ensure page is no longer in whitelist

**TEST 2: Test new buttons**
1. Navigate to a page and open the privacy Dashboard
1. Tap the _Manage Whitelist_ button and ensure whitelist page is launched
1. Ensure the "mp_mw" pixel fires
1. Go back
1. Tap the _Report Broken Site_ button and ensure that the broken site button in launched
1. Ensure the "mp_rb" pixel fires

**TEST 3: Test new pixels browsing pixels**
1. Browse to a site
1. From the browsing menu, add it to the whitelist and ensure that "mb_wla" fires
1. From the browsing menu, remove it from the whitelist and ensure that "mb_wlr" fires

**TEST 4: Test new settings pixel**
1. Open _Settings_
1. Tap _Privacy Protection Whitelist_
1. Ensure "ms_mw" fires


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
